spec/unit/block_score/actions/update_spec.rb

```diff
@@ -2,15 +2,41 @@
 
 RSpec.describe BlockScore::Actions::Update, vcr: true do
   describe '#save!' do
-    context 'when updating an existing candidate' do
-      subject(:candidate) { create(:candidate, name_first: 'John') }
+    subject(:candidate) { build(:candidate, name_first: 'John') }
+    context 'when updating an new candidate' do
       before do
-        candidate.name_first = 'Jane'
-        candidate.save!
+        expect(candidate.persisted?).to be false
+        expect(candidate.save!).to be true
+
+        expect(candidate.persisted?).to be true
+      end
+
+      it 'has the same attributes' do
+        expect(BlockScore::Candidate.retrieve(candidate.id)
+                 .attributes.reject { |key, _| key.equal?(:updated_at) })
+          .to eql(candidate.attributes.reject do |key, _|
+            key.equal?(:updated_at)
+          end)
       end
+    end
 
-      it { is_expected.to be_persisted }
-      its(:name_first) { is_expected.to eql 'Jane' }
+    context 'when changing first update candidate', vcr: { record: :all } do
+      it 'has the same attributes' do
+        expect(candidate.persisted?).to be false
+        expect(candidate.save!).to be true
```

Comment:
I agree this could work, although I do think the specs are a bit coarse grained. In a *true* unit test I would only test one specific behaviour. The return value would be tested separately from the state change behaviour.

Given these are integration tests its ok to test a few things in a specific `it` block given how expensive the tests are to execute. In a real unit test I would want a failure of a given `it` block to be self-evident -- something which is difficult when it tests multiple unrelated aspects of a method.